app/src/main/java/com/keylesspalace/tusky/EditProfileActivity.kt

```diff
@@ -306,4 +320,17 @@ class EditProfileActivity : BaseActivity(), Injectable {
         Log.w("EditProfileActivity", "failed to pick media", throwable)
         Snackbar.make(binding.avatarButton, R.string.error_media_upload_sending, Snackbar.LENGTH_LONG).show()
     }
+
+    private fun showUnsavedChangesDialog() = lifecycleScope.launch {
+        when (launchAlertDialog()) {
+            AlertDialog.BUTTON_POSITIVE -> save()
+            else -> finish()
+        }
+    }
+
+    private suspend fun launchAlertDialog() = AlertDialog.Builder(this)
```

Comment:
The `launchAlertDialog` is technical but does not say what it does: `showSaveDialog`?

(Which is the same name as the wrapping function...)

Reply:
I'll change it to launchSaveDialog() as it's a suspend function that makes the current "showUnsavedChangesDialog()" more readable without cramping the dialog logic there as well.

Reply:
is that OK?

Reply:
Yes.